File: Desafios em C#/1240 - Encaixa ou Não I.cs
using System;

//Developed by: @LucasMarcuzo

namespace _1240___Encaixa_ou_Não_I
{
    class Program
    {
        static void Main(string[] args)
        {   
            
            int casos = int.Parse(Console.ReadLine());

            for (int i = 0; i < casos; i++)
            {               
                string[] linha = Console.ReadLine().Split(' ');

                string primeiro = linha[0];
                string segundo = linha[1];
                
                int tam_primeiro = primeiro.Length;
                int tam_segundo = segundo.Length;

                int tam = tam_primeiro - tam_segundo;

                if (tam_primeiro < tam_segundo)
                {
                    Console.WriteLine("nao encaixa");
                }
                else
                {
                    string y = primeiro.Substring(tam, tam_segundo); 
                    bool x = y.Contains(segundo); 

                    if(x == true)
                    {
                        Console.WriteLine("encaixa"); 
                    }
                    else
                    {
                        Console.WriteLine("nao encaixa"); 
                    }                      
                }
            }
        }
    }
}

File: Desafios em JavaScript/1240 - Encaixa ou Não I.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

var casos = parseInt(lines.shift());

for (let i = 0; i < casos; i++)
{               
    var linha = lines.shift().split(" ");

    var primeiro = linha.shift();
    var segundo = linha.shift();
    
    primeiro = primeiro.replace("\r", "");
    segundo = segundo.replace("\r", "");

    var tam_primeiro = primeiro.length;
    var tam_segundo = segundo.length;


    var tam = tam_primeiro - tam_segundo;

    if (tam_primeiro < tam_segundo)
    {
       console.log("nao encaixa");
    }
    else
    {
        var y = primeiro.substring(tam); 
        var x = y.includes(segundo); 

        if(x === true)
        {
           console.log("encaixa"); 
        }
        else
        {
           console.log("nao encaixa"); 
        }                      
    }
}